### PR TITLE
Task 53432: Add Analytics metric for recordings in jitsi.

### PR DESCRIPTION
**Pull request by montassarhamadi**
Improvement: add two metrics of recorded calls (Recorded file size, uploading status) to module analytics. So in this implementation, I fired two events of the same type EVENT_CALL_RECORDED in success and failed cases of uploading of the recorded file, then I added the necessary instructions to process the data sent to the listener and add these two parameters in the recorded call case.

### CODE REVIEW

**Comment by hbenali** [comment]
Please rebase your branch and remote [SWF commit](https://github.com/exoplatform/web-conferencing/pull/96/commits/4c1848d7fcccb0e60236186ee8cc991d6f6a1e22)

File: services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java [hunk 1]
```diff
@@ -125,6 +127,10 @@ public void onEvent(Event<CallInfo, String> event) throws Exception {
     if (callDuration > 0) {
       statisticData.addParameter("callDuration", callDuration);
     }
+    if(statisticData.getOperation().equals("callRecorded") && info.containsKey("file_size")) {
+      statisticData.addParameter("recordedFileSize", info.get("file_size"));
+      statisticData.setStatus(info.get("upload_status").equals(WebConferencingService.STATUS_OK) ? StatisticData.StatisticStatus.OK : StatisticData.StatisticStatus.KO);
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java]:
```suggestion
```

Please use statisticData.setStatus, and do it outside of the if

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (rdenarie) [inline .vscode/settings.json]: Remove this file

File: pom.xml [hunk 3]
```diff
@@ -59,8 +59,11 @@
     <org.exoplatform.social.version>6.3.x-maintenance-SNAPSHOT</org.exoplatform.social.version>
     <org.exoplatform.platform-ui.version>6.3.x-maintenance-SNAPSHOT</org.exoplatform.platform-ui.version>
     <addon.exo.analytics.version>1.2.x-maintenance-SNAPSHOT</addon.exo.analytics.version>
+<<<<<<< HEAD
```
Comment (rdenarie) [inline pom.xml]: remove this

**Review by rdenarie — CHANGES_REQUESTED** [review]
Please check your PR, there is a lot of tags related to git merge.

File: services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java [hunk 4]
```diff
@@ -2107,18 +2095,17 @@ public void uploadFile(UploadFileInfo uploadInfo, HttpServletRequest request) th
...
-          LOG.info(metricMessage(uploadingUser, call, OPERATION_CALL_RECORDED, STATUS_OK, System.currentTimeMillis() - opStart, null, resource.getUploadedSize()));
-        } catch (InvalidCallException e) {
-          LOG.warn("Failed to build metric for " + OPERATION_CALL_RECORDED, e);
-        }
+        broacastCallEvent(EVENT_CALL_RECORDED, call, uploadingUser, resource.getUploadedSize(), STATUS_OK);
+        LOG.info(metricMessage(uploadingUser, call, OPERATION_CALL_RECORDED, STATUS_OK, System.currentTimeMillis() - opStart, null, resource.getUploadedSize()));
       } else {
         throw new UploadFileException("The file " + resource.getFileName() + " cannot be uploaded. Status: "
-            + resource.getStatus());
+                + resource.getStatus());
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java]: can you remove the added spaces ?

File: services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java [hunk 5]
```diff
@@ -3903,14 +3890,26 @@ protected String metricMessage(String userId,
...
   }
 
-  private void broacastCallEvent(String eventName, CallInfo call, String userId) {
+  private void broacastCallEvent(String eventName, CallInfo call, String userId, Double fileSize, String status) {
     try {
-      listenerService.broadcast(eventName, call, userId);
+      Map<String, String> info = new HashMap<>();
+      info.put("user_id", userId);
+      info.put("status", status);
+      if(eventName.equals(EVENT_CALL_RECORDED) && fileSize != null) {
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java]:
```suggestion
      if(fileSize != null) {
```

File: services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java [hunk 6]
```diff
@@ -125,6 +127,12 @@ public void onEvent(Event<CallInfo, String> event) throws Exception {
     if (callDuration > 0) {
       statisticData.addParameter("callDuration", callDuration);
     }
+    if (info.get("status") != null) {
+      statisticData.setStatus(info.get("status").equals(WebConferencingService.STATUS_OK) ? StatisticData.StatisticStatus.OK : StatisticData.StatisticStatus.KO);
+    }
+    if(statisticData.getOperation().equals("callRecorded") && info.containsKey("file_size")) {
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java]:
```suggestion
    if(info.containsKey("file_size")) {
```

File: services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java [hunk 7]
```diff
@@ -125,6 +127,12 @@ public void onEvent(Event<CallInfo, String> event) throws Exception {
...
+      statisticData.addParameter("recordedFileSize", info.get("file_size"));
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/webconferencing/listener/analytics/WebConferencingListener.java]:
```suggestion
      statisticData.addParameter("recordedFileSizeInMB", info.get("file_size"));
```

**Review by rdenarie — CHANGES_REQUESTED** [review]
Where did you add the i18n key for recordedFileSizeInMB ?

**Comment by montassarhamadi** [comment]
> Where did you add the i18n key for recordedFileSizeInMB ?

https://github.com/Meeds-io/analytics/pull/5

**Comment by rdenarie** [comment]
:+1:

**Review by rdenarie — APPROVED** [review]

**Pull request merged**